### ISSUE TITLE
 No module named geotiepoints.modisinterpolator

### ISSUE THREAD

**Issue by AlysonR** [pytroll/satpy#852]
when running this with composite set as some true_color or true_color_crefl it throws an error
ImportError: No module named geotiepoints.modisinterpolator
If you made geotiepoints pip able I could fix this. 
Code:
	
        #given a path, find all the two correct files within
	files = get_modis_filenames(path)
	#load the global scene using satpy
	#expects filenames to be geoloc first, then radiance path
	#filenames must be in specific format
	global_scene = Scene(reader = 'modis_l1b', filenames = files)
	#for future reference here are all available datasets
	available_datasets = global_scene.available_dataset_names()
	#want to create a cloudtop composite
	#for future reference all available composites
	available_composites = global_scene.available_composite_names()
	print available_composites
	#expects the composite to be in a list
	global_scene.load([composite])
	global_scene.show(composite)
	return

**Comment by djhoese** [comment]
This is likely due to those composites loading data from the modis_l1b reader that requires interpolation from 1km resolution to 500m or 250m resolution. That said, the package is installable from pip and conda (via conda-forge):

```
conda install -c conda-forge python-geotiepoints
```

```
pip install python-geotiepoints
```

**Comment by AlysonR** [comment]
It also requires ImportError: No module named pyorbital.astronomy
What should be pip installed to fix this? Is there a way to just have these installed with satpy to avoid these import errors?

**Comment by djhoese** [comment]
This is from the `pyorbital` package which can also be pip installed and conda installed. These packages are installed as dependencies with conda because of how easy it is to install the binary packages from conda. With pip we tried to avoid installing dependencies that weren't needed since they aren't needed by every reader, or composite, or writer. We tried to do this by making it possible by using "extras" with pip:

```
pip install satpy[modis_l1b]
```

But this can't easily cover all cases (dependencies for a writer, dependencies for a particular composite, etc).

Perhaps we should add geotiepoints and pyorbital as hard requirements. @mraspaud @sfinkens Thoughts?

In summary: This is a known headache of Satpy due to how modular it is and how many various dependencies could be involved in one workflow. I'm open to ideas for how we could better handle this (easily install or document dependencies for a workflow without installing a ton of dependencies when they aren't needed).

**Comment by sfinkens** [comment]
@djhoese Is it possible to add a `composites` extra to setup.py which installs common requirements for composites? Then you could do 
```
pip install satpy[modis_l1b,composites]
```
Or are the workflows too diverse for that?

**Comment by mraspaud** [comment]
@sfinkens that would be fine by me.

**Comment by djhoese** [comment]
How about pyorbital becomes a hard requirement? I don't think geotiepoints is used outside of the readers right now, right?

**Comment by mraspaud** [comment]
yes, I'm ok with this

**Comment by mraspaud** [comment]
yes geotiepoints is only for the readers atm

**Comment by sfinkens** [comment]
Me, too
